test/jit/test_symbolic_shape_analysis.py

```diff
@@ -0,0 +1,55 @@
+import torch
+from torch.testing._internal.jit_utils import JitTestCase
+
+from torch.testing import FileCheck
+from typing import List
+
+
+if __name__ == '__main__':
+    raise RuntimeError("This test file is not meant to be run directly, use:\n\n"
+                       "\tpython test/test_jit.py TESTNAME\n\n"
+                       "instead.")
+
+# XXX: still in prototype
+class TestSymbolicShapeAnalysis(JitTestCase):
+    def test_shape_analysis(self):
+        @torch.jit.script
+        def broadcast(a: List[int], b: List[int]):
```

Comment:
Since the test cases use `[None, None]` to set input sizes. Should the parameters of shape functions be `List[Optional[int]]`?

Reply:
Optional here is sugar around creating a new symbolic shape (i.e., not knowing anything about it). It's taken from the withSymbolicShapes API. By the time the function is invoked  the Nones get bound to an integer